src/authorized_principals/authorized_principals.cc

```diff
@@ -90,6 +90,10 @@ int main(int argc, char* argv[]) {
   opts.fingerprint = fingerprint;
   opts.fp_len = fp_len;
 
+  if (cloud_run) {
```

Comment:
Call me a functional programming purist, but IMO it's usually a bad code smell to reassign a variable outside of a loop. 

You've already got the `principal` variable as a separate variable. Rather than replace `user_name` blindly here, please check that it's as expected (in this case, you expect it to always be "root," right?) and then call `AuthorizeUser` on the `principal` (i.e. call `AuthorizeUser(principal, opts, *user_response, cloud_run)` instead.